src/Agent.Worker/JobRunner.cs

```diff
@@ -10,12 +11,12 @@ namespace Microsoft.VisualStudio.Services.Agent.Worker
     [ServiceLocator(Default = typeof(JobRunner))]
     public interface IJobRunner : IAgentService
     {
-        Task<TaskResult> RunAsync(JobRequestMessage message);
+        Task<TaskResult> RunAsync(JobRequestMessage message, CancellationToken jobRequestCancellationToken);
     }
 
     public sealed class JobRunner : AgentService, IJobRunner
     {
-        public async Task<TaskResult> RunAsync(JobRequestMessage message)
+        public async Task<TaskResult> RunAsync(JobRequestMessage message, CancellationToken jobRequestCancellationToken)
```

Comment:
We need to catch OperationCancellationException in jobRunner.RunAsync in order to set the jobContext.Result to TaskResult.Canceled. We have some duplication of code where we do "TaskResult.Failed": we have catch for stepsRunner.RunAsync, taskManager.DownloadAsync, a global catch for the entire jobRunner.RunAsync and they all do "jobContext.Result = TaskResult.Failed;". Also the StepRunner inside catches some exceptions and sets jobContext.Result = TaskResult.Failed, which could be accomplished if it lets the exception bubble to jobRunner.